### ISSUE TITLE
[CT-662] [Feature] Allow the configuration of the timezone for timestamps in event logging

### ISSUE THREAD

**Issue by Stefan-Rann** [dbt-labs/dbt-core#5267]
### Is this your first time opening an issue?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)

### Describe the Feature

Currently the timestamp in event logging is always output in utc.
It would be helpful to be able to set the time zone via a configuration or parameter (either local or any other time zone).

### Describe alternatives you've considered

I have not found any alternatives. One has to live with utc.

### Who will this benefit?

This would make it easier to evaluate the logs and help to avoid confusion. 
Additionally the evaluation of dbt logs in combination with other logs using lokal or any other timezone would be easier.

**Comment by gshank** [comment]
This sounds like a reasonable request. We're going to be looking at logging updates later in this quarter. I'm not sure if it would get onto the schedule or not, but we can take a look.

**Comment by jtcohen6** [comment]
Reopening because we just got a duplicate opened in https://github.com/dbt-labs/dbt-core/issues/6319

> Things I'm curious about: How would you want to configure this? Should there only be two options (local system time versus UTC)?

**Comment by ahuiiiiiii** [comment]
> Reopening because we just got a duplicate opened in #6319
> 
> > Things I'm curious about: How would you want to configure this? Should there only be two options (local system time versus UTC)?

can i add an option that set zonetime in dbt-core project configuration file?

**Comment by ahuiiiiiii** [comment]
sorry spelling error above should be timezone

**Comment by Stefan-Rann** [comment]
> can i add an option that set zonetime in dbt-core project configuration file?

Yes, a project level setting would be cool to set the timezone by name.
Additionally the option "local" would be great to use the system timezone without the need to explicitly define the timezone.

**Comment by Gunnnn** [comment]
Please! I see timestamps in log in UTC and its annoying =(

**Comment by Frunkh** [comment]
Atleast change 
`record.extra[self.name] = datetime.utcnow().isoformat()`
to 
`record.extra[self.name] = datetime.now().isoformat()`

in logger.py
So it respects the timezone of the system.. this is really annoying.

**Comment by sgaraycoa-de** [comment]
Adding my vote in favor of this issue

**Comment by jtcohen6** [comment]
Putting aside, for the moment, the question of creating a standalone configuration for this — I'd be supportive of switching from `datetime.utcnow()` to `datetime.now()`, so that the user's local system timezone is what's displayed in the logs. My sense is that most remotely-running orchestration systems would be in UTC, so it wouldn't be a change there.

Timestamp Doug (@dbeatty10) - does that proposed switch give you any trepidation?

**Comment by nevdelap** [comment]
Just having it log in the system's timezone would be great. I'm lucky enough to be at UTC+10 so it is only mildly annoying, but still strange that it would be that way.

**Comment by jim256** [comment]
I'd also really like to see two options (local system time versus UTC) rather than always defaulting to UTC only.

**Comment by PORXavierM** [comment]
As somebody find solution for this? is a problem from python or dbt himself?

**Comment by dbeatty10** [comment]
@PORXavierM the timestamps are coming from the logging of `dbt`.

There is some of amount of configurability of the timestamps via the [`LOG_FORMAT`](https://docs.getdbt.com/reference/global-configs/logs#log-formatting) and [`LOG_FORMAT_FILE`](https://docs.getdbt.com/reference/global-configs/logs#log-formatting) configs. Here's a brief summary:

| Log format | Time zone              | Aware vs. naive | Precision   |
|------------|------------------------|-----------------|-------------|
| `text`     | UTC                    | naive timestamp | second      |
| `debug`    | system local time zone | naive timestamp | microsecond |
| `json`     | UTC                    | aware timestamp | microsecond |

There's some more nitty-gritty code details discussed [here](https://github.com/dbt-labs/docs.getdbt.com/pull/5578).
